cmd/vela-makisu/build.go

```diff
@@ -4,15 +4,561 @@
 
+	// verify tag are provided
+	if len(b.Context) == 0 {
+		return fmt.Errorf("no build context provided")
+	}
+
+	// verify tag are provided
+	if len(b.Tag) == 0 {
+		return fmt.Errorf("no build tag provided")
+	}
+
 	return nil
 }
+
+// Flags formats and outputs the flags for
+// configuring a Docker.
+func (d *Docker) Flags() []string {
+	// variable to store flags for command
+	var flags []string
+
+	// check if Host is provided
+	if len(d.Host) > 0 {
+		// add flag for Host from provided build command
+		flags = append(flags, fmt.Sprintf("--docker-host=%s", d.Host))
+	}
+
+	// check if Scheme is provided
+	if len(d.Scheme) > 0 {
+		// add flag for Scheme from provided build command
+		flags = append(flags, fmt.Sprintf("--docker-scheme=%s", d.Scheme))
+	}
+
+	// check if Version is provided
+	if len(d.Version) > 0 {
+		// add flag for Version from provided build command
+		flags = append(flags, fmt.Sprintf("--docker-version=%s", d.Version))
+	}
+
+	return flags
+}
+
+// Flags formats and outputs the flags for
+// configuring a http cache.
+func (h *HTTPCache) Flags() []string {
+	// variable to store flags for command
+	var flags []string
+
+	// check if Addr is provided
+	if len(h.Addr) > 0 {
+		// add flag for Addr from provided build command
+		flags = append(flags, fmt.Sprintf("--http-cache-addr=%s", h.Addr))
+	}
+
+	// check if Headers is provided
+	if len(h.Headers) > 0 {
+		var args string
+		for _, arg := range h.Headers {
+			args += fmt.Sprintf(" %s", arg)
+		}
+		// add flag for BuildArgs from provided build command
+		flags = append(flags, fmt.Sprintf("--http-cache-header=%s", strings.TrimPrefix(args, " ")))
+	}
+
+	return flags
+}
+
+// Flags formats and outputs the flags for
+// configuring a redis cache.
+func (r *RedisCache) Flags() ([]string, error) {
+	// variable to store flags for command
+	var flags []string
+
+	// check if Addr is provided
+	if len(r.Addr) > 0 {
+		// add flag for Addr from provided build command
+		flags = append(flags, fmt.Sprintf("--redis-cache-addr=%s", r.Addr))
+	}
+
+	// check if Password is provided
+	if len(r.Password) > 0 {
+		// add flag for Password from provided build command
+		flags = append(flags, fmt.Sprintf("--redis-cache-password=%s", r.Password))
+	}
+
+	// check if TTL is provided
+	if len(r.TTL) > 0 {
+		duration, err := time.ParseDuration(r.TTL)
+		if err != nil {
+			return nil, err
+		}
+
+		// check if TTL is valid duration
+		if !isDurationZero(duration) {
+			// add flag for TTL from provided build command
+			flags = append(flags, fmt.Sprintf("--redis-cache-ttl=%s", duration))
+		}
+	}
+
+	return flags, nil
+}
+
+// helper function to check if the time in duration
+// is the zero value.
+func isDurationZero(t time.Duration) bool {
+	zero, err := time.ParseDuration("0")
+	if err != nil {
+		logrus.Error(err)
+		return false
+	}
+
+	// if the duration is zero return true
+	if strings.EqualFold(t.String(), zero.String()) {
+		return true
+	}
+
+	return false
+}
```

Comment:
I think we can remove this whole function since `time.Duration` can be compared against `0` since it has a type of `int64` 👍

Reply:
Responded above